File: Day274_2025/Q274_Leetcode.cpp
class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        // Q. Water Bottles
        int totalBottles = numBottles;
        while (numBottles >= numExchange) {
            int emptyBottles = numBottles / numExchange;
            totalBottles += emptyBottles;
            numBottles = emptyBottles + (numBottles % numExchange);
        }
        return totalBottles;
    }
};

File: Day274_2025/Q274_Leetcode.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        totalBottles = numBottles
        while numBottles >= numExchange:
            emptyBottles = numBottles // numExchange
            totalBottles += emptyBottles
            numBottles = emptyBottles + (numBottles % numExchange)
        return totalBottles
